src/lasdi/latent_space.py

```diff
@@ -1,175 +1,482 @@
-import torch
-import numpy as np
+# -------------------------------------------------------------------------------------------------
+# Imports and setup
+# -------------------------------------------------------------------------------------------------
+
+import  torch
+import  numpy       as      np
+
+from   .physics     import  Physics
 
 # activation dict
-act_dict = {'ELU': torch.nn.ELU,
-            'hardshrink': torch.nn.Hardshrink,
-            'hardsigmoid': torch.nn.Hardsigmoid,
-            'hardtanh': torch.nn.Hardtanh,
-            'hardswish': torch.nn.Hardswish,
-            'leakyReLU': torch.nn.LeakyReLU,
-            'logsigmoid': torch.nn.LogSigmoid,
-            'multihead': torch.nn.MultiheadAttention,
-            'PReLU': torch.nn.PReLU,
-            'ReLU': torch.nn.ReLU,
-            'ReLU6': torch.nn.ReLU6,
-            'RReLU': torch.nn.RReLU,
-            'SELU': torch.nn.SELU,
-            'CELU': torch.nn.CELU,
-            'GELU': torch.nn.GELU,
-            'sigmoid': torch.nn.Sigmoid,
-            'SiLU': torch.nn.SiLU,
-            'mish': torch.nn.Mish,
-            'softplus': torch.nn.Softplus,
-            'softshrink': torch.nn.Softshrink,
-            'tanh': torch.nn.Tanh,
-            'tanhshrink': torch.nn.Tanhshrink,
-            'threshold': torch.nn.Threshold,
-            }
-
-def initial_condition_latent(param_grid, physics, autoencoder):
-
-    '''
-
-    Outputs the initial condition in the latent space: Z0 = encoder(U0)
-
-    '''
-
-    n_param = param_grid.shape[0]
-    Z0 = []
-
-    sol_shape = [1, 1] + physics.qgrid_size
+act_dict = {'ELU'           : torch.nn.ELU,
+            'hardshrink'    : torch.nn.Hardshrink,
+            'hardsigmoid'   : torch.nn.Hardsigmoid,
+            'hardtanh'      : torch.nn.Hardtanh,
+            'hardswish'     : torch.nn.Hardswish,
+            'leakyReLU'     : torch.nn.LeakyReLU,
+            'logsigmoid'    : torch.nn.LogSigmoid,
+            'PReLU'         : torch.nn.PReLU,
+            'ReLU'          : torch.nn.ReLU,
+            'ReLU6'         : torch.nn.ReLU6,
+            'RReLU'         : torch.nn.RReLU,
+            'SELU'          : torch.nn.SELU,
+            'CELU'          : torch.nn.CELU,
+            'GELU'          : torch.nn.GELU,
+            'sigmoid'       : torch.nn.Sigmoid,
+            'SiLU'          : torch.nn.SiLU,
+            'mish'          : torch.nn.Mish,
+            'softplus'      : torch.nn.Softplus,
+            'softshrink'    : torch.nn.Softshrink,
+            'tanh'          : torch.nn.Tanh,
+            'tanhshrink'    : torch.nn.Tanhshrink,
+            'threshold'     : torch.nn.Threshold,}
+
+
+
+# -------------------------------------------------------------------------------------------------
+# initial_conditions_latent function
```

Comment:
In order to be parsed in API document as one-line description, this line can be moved to line 43 with an additional empty line. Something like below:

```
def initial_condition_latent(param_grid     : np.ndarray, 
                             physics        : Physics, 
                             autoencoder    : torch.nn.Module) -> list[np.ndarray]:
    """initial_conditions_latent function

    This function maps a set of initial conditions for the fom to initial conditions for the
```